realm/realm-library/src/androidTest/java/io/realm/RealmObjectSchemaTests.java

```diff
@@ -0,0 +1,586 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.realm;
+
+import android.test.AndroidTestCase;
+
+import java.util.Date;
+import java.util.Set;
+
+import io.realm.entities.AllJavaTypes;
+
+public class RealmObjectSchemaTests extends AndroidTestCase {
+
+    private RealmObjectSchema DOG_SCHEMA;
+    private DynamicRealm realm;
+    private RealmObjectSchema schema;
+    private RealmSchema realmSchema;
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        RealmConfiguration realmConfig = new RealmConfiguration.Builder(getContext()).build();
+        Realm.deleteRealm(realmConfig);
+        Realm.getInstance(realmConfig).close(); // Create Schema
+        realm = DynamicRealm.getInstance(realmConfig);
+        realmSchema = realm.getSchema();
+        DOG_SCHEMA = realmSchema.get("Dog");
+        realm.beginTransaction();
+        schema = realmSchema.create("NewClass");
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        super.tearDown();
```

Comment:
It can probably no practical impact, but shouldn't `super.tearDown()` be at the end?